pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -214,12 +214,17 @@ int32_t free_array(t_ndarray arr)
 
 int32_t free_pointer(t_ndarray arr)
 {
-    if (arr.is_view == false || arr.shape == NULL)
+    if (arr.shape == NULL)
         return (0);
     free(arr.shape);
     arr.shape = NULL;
     free(arr.strides);
     arr.strides = NULL;
+    if (arr.is_view == false)

```

Comment:
Can an object have `is_view ==false` and be a pointer?

Reply:
I suspect you are following confusing Fortran implementations needed for the wrapper. I'm not sure this is needed in C. (or in Fortran outside of the wrapper)

Reply:
it's @bauom idea 😅 after working with pointer, I think we don't need it now since we don't need the returned object to be a pointer, I forget why needed it 🤔